tfx/components/example_gen/base_example_gen_executor_test.py

```diff
@@ -124,58 +130,46 @@ def testDoInputSplit(self):
-    self._exec_properties[utils.OUTPUT_CONFIG_KEY] = json_format.MessageToJson(
-        example_gen_pb2.Output(
-            split_config=example_gen_pb2.SplitConfig(splits=[
-                example_gen_pb2.SplitConfig.Split(name='train', hash_buckets=2),
-                example_gen_pb2.SplitConfig.Split(name='eval', hash_buckets=1)
-            ])))
-
-    # Run executor.
-    example_gen = TestExampleGenExecutor()
-    example_gen.Do({}, self._output_dict, self._exec_properties)
+  def testDoInputSplit(self):
+    # Create exec proterties for input split.
+    exec_properties = {
+        utils.INPUT_CONFIG_KEY:
+            json_format.MessageToJson(
+                example_gen_pb2.Input(splits=[
+                    example_gen_pb2.Input.Split(
+                        name='train', pattern='train/*'),
+                    example_gen_pb2.Input.Split(name='eval', pattern='eval/*')
+                ]),
+                preserving_proto_field_name=True),
+        utils.OUTPUT_CONFIG_KEY:
+            json_format.MessageToJson(
+                example_gen_pb2.Output(), preserving_proto_field_name=True)
+    }
 
-    # Check example gen outputs.
-    self.assertTrue(tf.io.gfile.exists(self._train_output_file))
-    self.assertTrue(tf.io.gfile.exists(self._eval_output_file))
+    self._testDo(exec_properties)
 
-    # Output split ratio: train:eval=2:1.
-    self.assertGreater(
-        tf.io.gfile.GFile(self._train_output_file).size(),
-        tf.io.gfile.GFile(self._eval_output_file).size())
+  def testDoOutputSplit(self):
+    self._testDo(self._exec_properties)
 
   def testDoOutputSplitWithProto(self):
-    # Add output config to exec proterties.
-    self._exec_properties[utils.OUTPUT_CONFIG_KEY] = json_format.MessageToJson(
-        example_gen_pb2.Output(
-            split_config=example_gen_pb2.SplitConfig(splits=[
-                example_gen_pb2.SplitConfig.Split(name='train', hash_buckets=2),
-                example_gen_pb2.SplitConfig.Split(name='eval', hash_buckets=1)
-            ])))
+    # Update exec proterties.
     self._exec_properties['format_proto'] = True
 
-    # Run executor.
-    example_gen = TestExampleGenExecutor()
-    example_gen.Do({}, self._output_dict, self._exec_properties)
+    self._testDo(self._exec_properties)
 
-    # Check example gen outputs.
-    self.assertTrue(tf.io.gfile.exists(self._train_output_file))
-    self.assertTrue(tf.io.gfile.exists(self._eval_output_file))
+  def testDoOutputSplitWithSequenceExample(self):
+    # Update exec proterties.
+    self._exec_properties['sequence_example'] = True
 
-    # Output split ratio: train:eval=2:1.
-    self.assertGreater(
-        tf.io.gfile.GFile(self._train_output_file).size(),
-        tf.io.gfile.GFile(self._eval_output_file).size())
+    self._testDo(self._exec_properties)
 
   def testFeatureBasedPartition(self):
-    # Add output config to exec proterties.
+    # Update output config in exec proterties.
```

Comment:
shall we add a test to test feature based partition with sequence example?

Reply:
yes sure, i added a test for this